framework/packages/abstract-interface/src/account/mod.rs

```diff
@@ -328,6 +336,119 @@ impl<Chain: CwEnv> AbstractAccount<Chain> {
         Ok(migrated)
     }
 
+    pub fn upgrade(&self) -> Result<(), AbstractInterfaceError> {
+        let mut one_migration_was_successful = false;
+
+        // We upgrade the manager to the latest version through all the versions
+        loop {
+            if self.upgrade_next_module_version(MANAGER).is_err() {
+                break;
+            }
+            one_migration_was_successful = true;
+        }
+
+        loop {
+            if self.upgrade_next_module_version(PROXY).is_err() {
+                break;
+            }
+            one_migration_was_successful = true;
+        }
+
+        if !one_migration_was_successful {
```

Comment:
Why do we want to error if the upgrade wasn't needed? I'd prefer throwing Options here so the user can decide what to do more easily.

Reply:
Yeah, agree, returning `one_migration_was_successful` would be nicer here